src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -410,9 +407,9 @@ private SslHandler createHandler(SSLEngine engine, boolean client) {
           break;
         }
       }
-    } else if (verifyHost) {
+    } else if (!endpointIdentificationAlgorithm.isEmpty()) {
```

Comment:
Perhaps endpointIdentificationAlgorithm should be an enum instead? Would remove both null and invalid option values.